cfpb/cfgov-refresh
# Mortgage performance: Automate downloads page

Our data pipeline will keep track of download files with each quarterly
update, so we can automate the download archives page with a block that
has access to mortgage performance metadata.

This PR adds a MortgageDataDownloads block and template to render the
download data produced by each data update.

## Testing
To see the archive page in action locally:
- Make sure you have a recent db dump loaded so you don't have to create our project pages. You'll get a data downloads page, but you'll be creating another one.
- Run Django migration to pick up the new block:
```
python cfgov/manage.py migrate --noinput
```

- Download data with sample archive metadata included:
    - http://files.consumerfinance.gov.s3.amazonaws.com/data/mortgage-performance/initial_data/mortgagedata_archive_data.sql.gz
- load the data:

```
gunzip -c ~/Downloads/mortgagedata_archive_data.sql.gz | mysql -uroot v1
```

- In the Wagtail admin page Explorer, navigate to `CFGov > Data & Research > Mortgage Performance Trends` and add a child page, choosing `Mortgage performance page`. You'll be creating a duplicate downloads page to pull in automated archive listings, so give a fake title like "Download the data2"

Under the new page's content section, add a Mortgage Downloads Block, and check the "Show archives" box. Then publish the page.

When you view the new page live, you should see two sections, One listing downloads for the most recent data and another section below listing two sets of archival downloads. Links should all work. Since all the archival test files were created at once, they all have September 2017 as the pub date, but in the future those dates would reflect when quarterly data updates occurred.

The archive block should look like this:

<img width="679" alt="archive_block" src="https://user-images.githubusercontent.com/515885/30594952-c46c9ca2-9d1d-11e7-91c0-19d9fd0399de.png">
